cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -1097,6 +1097,7 @@ impl<I: VCodeInst> VCode<I> {
             }
         }
 
+        self.monotonize_inst_offsets(&mut inst_offsets[..], func_body_len);
         let value_labels_ranges =
             self.compute_value_labels_ranges(regalloc, &inst_offsets[..], func_body_len);
```

Comment:
It is interesting to note that `optimize_branches` is invoked after this code one last time and can nominally modify the current buffer pointer, make `func_body_len` stale, etc. At the same time, for it to remove a branch, that branch needs to point past the end of the function, which doesn't seem possible?